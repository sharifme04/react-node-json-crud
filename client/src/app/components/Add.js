import React, { useState } from "react";
import { useNavigate } from "react-router-dom";
import { useDispatch } from "react-redux";
import Form from "./Form";
import { addToDo } from "../actions/todo";

const Add = () => {
  let navigate = useNavigate();
  const dispatch = useDispatch();
  const [values, setValues] = useState({
    title: "",
    description: "",
  });

  const handleChange = (event) => {
    const { name, value } = event.target;
    setValues({ ...values, [name]: value });
  };

  const handleSubmit = (event) => {
    event.preventDefault();
    dispatch(addToDo(values));
    navigate("/");
  };

  return (
    <Form
      values={values}
      handleChange={handleChange}
      handleSubmit={handleSubmit}
    />
  );
};

export default Add;
