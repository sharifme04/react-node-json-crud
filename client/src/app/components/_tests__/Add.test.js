import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import { MemoryRouter } from "react-router-dom";
import Add from "../Add";

const mockStore = configureMockStore();
const store = mockStore({});

test("snapshot form renders", () => {
  const component = renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <Add />
      </MemoryRouter>
    </Provider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
