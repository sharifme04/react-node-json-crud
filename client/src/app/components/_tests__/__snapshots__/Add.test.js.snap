// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot form renders 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <label>
      Title:
    </label>
    <input
      className="form-control"
      name="title"
      onChange={[Function]}
      type="label"
      value=""
    />
  </div>
  <div
    className="form-group mt-3"
  >
    <label>
      Description:
    </label>
    <input
      className="form-control"
      name="description"
      onChange={[Function]}
      type="label"
      value=""
    />
  </div>
  <div
    className="mt-4"
  />
  <input
    className="btn btn-primary"
    type="submit"
    value="Add"
  />
</form>
`;
